crates/bevy_ecs/src/schedule/config.rs

```diff
@@ -36,24 +36,68 @@ fn ambiguous_with(graph_info: &mut GraphInfo, set: InternedSystemSet) {
     }
 }
 
+/// Stores data to differentiate different Node types
+pub trait NodeType {
+    /// Used in `NodeConfig`
+    type Metadata;
+    /// Used in `NodeConfigs`
+    type GroupMetadata;
+
+    /// Initialize `NodeConfig` from `NodeType`
```

Comment:
```suggestion
    /// Initializes a configuration from this node.
```
We can use normal english here since `NodeConfig` is already given as the return type, and this is a trait method on `Nodetype`.